oeps/processes/oep-0055/decisions/0003-readme-specification.rst

```diff
@@ -0,0 +1,183 @@
+0003 README specification
+#########################
+
+Status
+******
+
+** Accepted **
+
+Context
+*******
+
+`OEP-55`_ states that one of the jobs of a maintainer is te maintain an
+up-to-date and useful readme. While it laid out some possibilites about what
+should go into a README, it did not go into much detail.
+
+Decision
+********
+
+We will define a specification that READMEs in the open edx project must
+follow.  By having a single definition, we can make it easier for people to
+navigate between projects and find the most important information quickly.
+
+Specification
+*************
+
+The readme of a well-maintained repo should contain at least the following
+sections.  Ideally in this exact order.
+
+* **Title** - The name of the component.
+
+* **Badges** - A set of `badges <https://github.com/badges/shields>`_ that
+  quickly let visitors review the health of the project.
+
+	 * The badges section should contain at minimum the following badges:
+
+		  * **License** - Indicating the license of the repo.
+
+		  * **Maintenance Status** - One of ``maintained``, ``deprecated``, ``unsupported``, ``experimental``.
+
+		  * **CI Status** - The status of CI testing on the main branch.
+
+		  * **Package Published Status** - If the component is a library, then a badge indicating whether publishes to the relevant package index are passing. Eg. Has the latest version been successfully published to PyPI or NPM?
+
+
+* **Purpose** - A paragraph or two summarizing the purpose of this component.
+
+* **Getting Started** - A section to indicate how you can begin development on
+  the new component.
+
+* **Getting Help** - A section to indicate where and how the user can get help
+  with the component.  It should link to relevant docs as well as places where
+  the users can get help with from other humans.
+
+* **How to Contribute** - A section to indicate how new users can contribute to
```

Comment:
Do we need to have a status or similar around whether contributions are desired and/or will be accepted?

Reply:
I was considering renaming this section to `Contributing` and then we could let maintainers decide what level of contributions this component is accepting.  If we did that, I could imagine us having a status for the level of contribution accepted.  Possible values being:

* fixes only - no new features are being accepted at this point, but fixes and maintenance updates welcome.
* new-features accepted - please discuss your new ideas with the maintainers before you write code to increase the changes that features are accepted.
* security only - no changes accepted except for security fixes.

@e0d what do you think, is it worth updating this?

Reply:
That makes sense to me.  Though I think that we should emphasize the following:
1. contributions should always start with an issue, before code
2. we prefer maintainers who accept contributions